data/transactions/logic/assembler.go

```diff
@@ -1302,6 +1355,9 @@ func (ops *OpStream) assemble(text string) error {
 		return ops.errorf("Can not assemble version %d", ops.Version)
 	}
 	scanner := bufio.NewScanner(fin)
+	for i := range ops.known.scratchSpace {
```

Comment:
@iten-alg I opened a minor, optional child PR:  https://github.com/iten-alg/go-algorand/pull/9.  Can you take a look to see if you feel it's worth pulling in?

If you prefer as is, feel welcomed to resolve the thread as is.

Reply:
That is definitely much cleaner. Merged.